sherpa/fit.py

```diff
@@ -1163,27 +1156,34 @@ def __init__(self,
 
         # Set up an IterFit object, so that the user can select
         # an iterative fitting option.
-        self._iterfit = IterFit(self.data, self.model, statobj,
-                                self.method, iopts)
-
-        # We need to set the statistic *after* creating the _iterfit
-        # attribute
-        #
-        self.stat = statobj
+        self._iterfit = IterFit(self.data,
+                                self.model,
+                                Chi2Gehrels() if stat is None else stat,
+                                LevMar() if method is None else method,
+                                iopts)
 
         super().__init__()
 
     @property
     def stat(self) -> Stat:
-        """Return the statistic value"""
-        return self._stat
+        """Return the statistic object"""
+        return self._iterfit.stat
 
     @stat.setter
     def stat(self, stat: Stat) -> None:
         """Ensure that we use a consistent stat object."""
-        self._stat = stat
         self._iterfit.stat = stat
 
+    @property
+    def method(self) -> OptMethod:
+        """Return the method object"""
+        return self._iterfit.method
+
+    @method.setter
+    def method(self, method: OptMethod) -> None:
+        """Ensure that we use a consistent method object."""
```

Comment:
So, I will note that the same discussion is relevant for `stat` as it is for `method`, and only the latter has been added here (i.e. the existing code already shows the thing I am about to asl about, and I probably wrote the original code).

We have a docstring on both the `@property` and `@property.setter` functions. Only the `@property` is used, so should we change them (both `stat` and `method`) to something like

```
"""Set or get the statistic/method object.""
```

and then we can remove the docstring from the `@foo.setter` version?